core/src/main/java/org/springframework/security/oauth2/server/authorization/web/OAuth2ClientAuthenticationFilter.java

```diff
@@ -16,25 +16,103 @@
 package org.springframework.security.oauth2.server.authorization.web;
 
 import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.oauth2.server.authorization.authentication.OAuth2ClientAuthenticationToken;
+import org.springframework.util.Assert;
 import org.springframework.web.filter.OncePerRequestFilter;
-
 import javax.servlet.FilterChain;
 import javax.servlet.ServletException;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.io.IOException;
 
 /**
+ * Processes a HTTP request's BASIC authorization headers, putting the result into the
```

Comment:
Please remove this javadoc that came from `BasicAuthenticationFilter` as not all details apply to this filter. For example, this `Filter` will support multiple authentication schemes (not just basic) via a composite `AuthenticationConverter`.